openhtf/plugs/frontline/bluetooth_sniffer.py

```diff
@@ -0,0 +1,277 @@
+# Copyright 2017 Google Inc. All Rights Reserved.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""Classes for controlling Frontline bluetooth sniffers.
+
+This module provides control over a variety of Frontline (now owned by Teledyne
+Lecroy) bluetooth sniffers. For full details on Frontline's sniffer control
+protocol, which this module partially implements, see the reference document:
+
+http://www.fte.com/docs/usermanuals/comprobe%20automation%20server%20protocol.pdf
+"""
+
+import argparse
+import contextlib
+import datetime
+import logging
+import os
+import socket
+import time
+
+from openhtf.util import files
+
+DEFAULT_AUTOMATION_PORT = 22901
+DEFAULT_HTTP_PORT = 55455
+REMOTE_BIN_PATH = (r'C:\Program Files (x86)\Frontline Test System II'
+                   r'\Frontline\Executables\Core')
+REMOTE_LOG_PATH = r'C:\ble_logs'
+VERSION = 'SNIFF00'
+
+_LOG = logging.getLogger(__name__)
+
+
+class SnifferError(Exception):
+  """Raised when a BLE sniffer command fails."""
+  def __init__(self, command, response):
+    super(SnifferError, self).__init__(
+        'Command "{}" failed with response "{}"'.format(command, response))
+    self.response = response
+
+
+class Sniffer(object):
+  """Class for controlling a Frontline bluetooth sniffer via Telnet.
+
+  Args:
+    host: Hostname or IP address of the sniffer's automation server.
+    port: TCP port fort the sniffer's automation server.
+    log_path: Path on the sniffer's automation server at which to save logs.
+  """
+  def __init__(self,
+               host='localhost',
+               port=DEFAULT_AUTOMATION_PORT,
+               log_path=REMOTE_LOG_PATH):
+    self._address = host, port
+    self._log_path = log_path
+
+    self.capture_id = None
+    self.capture_start_time = None
+    self._files = []
+    self._socket = None
+
+  def _connect(self):
+    """Open a connection to the sniffer's automation server."""
+    _LOG.info('Connecting to sniffer server at %s:%d', *self._address)
+    assert self._socket is None
+    self._socket = socket.create_connection(self._address, 10)
+
+    _LOG.info('Stopping any previously running sniffer captures')
+    self._socket.settimeout(16)
+    try:
+      self._sniffer_command('Stop FTS')
+    except (SnifferError, socket.timeout):
+      pass
+
+    self._socket.settimeout(None)
+
+  def _sniffer_command(self, command, *args):
+    """Send command to the sniffer's automation server and wait for response."""
```

Comment:
maybe a little more description about the 'SUCCEEDED' response check/mention SnifferError in the docstring

Reply:
Done.